pygmt/base_plotting.py

```diff
@@ -621,7 +621,7 @@ def image(self, imagefile, **kwargs):
     @fmt_docstring
     @use_alias(R="region", J="projection", D="position", F="box")
     @kwargs_to_strings(R="sequence")
-    def legend(self, spec=None, **kwargs):
+    def legend(self, spec=None, position="JTR+jTR", **kwargs):
```

Comment:
Okay, adding an offset makes sense. What about a frame? The docs you linked say "We draw a white rectangular panel with a 1 point black outline..."

This would be more consistent with MATLAB / Matplotlib styling as well, for what it's worth.

Reply:
True, we might as well go all the way. Would be nice to see a side by side comparison of PyGMT and GMT's auto-legend actually, if it's not too much work of course.

Reply:
GMT code:
```fish
#!/usr/local/bin/fish

gmt begin plot pdf
  gmt plot -R0/6/3/6 -Jx1i -B @Table_5_11.txt -Sc0.15i -Glightgreen -Wfaint -lApples
  gmt plot @Table_5_11.txt -W1.5p,gray -l"My Lines"
  gmt plot @Table_5_11.txt -St0.15i -Gorange -lOranges
gmt end show
```

GMT output:

![gmt](https://user-images.githubusercontent.com/38269494/68377823-86666100-00f7-11ea-8cc4-168d7ac57ca8.png)

Reply:
PyGMT code (default legend position but with 0.2 cm offset, frame with white fill and 1 pt pen):
```python
import pygmt

fig = pygmt.Figure()

fig.basemap(projection="x1i", region=[0, 6, 3, 6], frame=True)

fig.plot(
    data="@Table_5_11.txt",
    style="c0.15i",
    color="lightgreen",
    pen="faint",
    label="Apples",
)
fig.plot(data="@Table_5_11.txt", pen="1.5p,gray", label='"My lines"')
fig.plot(data="@Table_5_11.txt", style="t0.15i", color="orange", label="Oranges")

fig.legend(position='JTR+jTR+o0.2c', box='+gwhite+p1p')

fig.show(method='external')
```

PyGMT plot:

![pygmt](https://user-images.githubusercontent.com/38269494/68378188-215f3b00-00f8-11ea-97cb-c784c8161390.png)

Reply:
Looks good, but does that mean we need to set a default `box` argument as well...

Reply:
Yup!

Reply:
Right, but maybe only set the box argument if **both** the `position (D)` and `box (F)` arguments are not given, otherwise the user might get something he/she doesn't want.

Reply:
Just found that out when I ran my tests! Good point!

Reply:
@weiji14 let me know if this is what you were imagining. I ran through the logic in my head and I think it checks out...